src/android/Geolocation.java

```diff
@@ -217,31 +229,40 @@ private void clearWatch(JSONArray args, CallbackContext callbackContext) {
     @Override
     public void onLocationResultSuccess(LocationContext locationContext, LocationResult locationResult) {
         for (Location location : locationResult.getLocations()) {
-            try {
-                JSONObject locationObject = LocationUtils.locationToJSON(location);
-                PluginResult result = new PluginResult(PluginResult.Status.OK, locationObject);
+            sendLocationResultSuccess(locationContext, location);
+        }
+    }
 
-                if (locationContext.getType() == LocationContext.Type.UPDATE) {
-                    result.setKeepCallback(true);
-                }
-                else {
-                    locationContexts.delete(locationContext.getId());
-                }
+    @Override
+    public void onLocationGPSResultSuccess(LocationContext locationContext, Location locationResult) {
+        sendLocationResultSuccess(locationContext, locationResult);
+    }
 
-                locationContext.getCallbackContext().sendPluginResult(result);
+    private void sendLocationResultSuccess(LocationContext locationContext, Location locationResult) {
```

Comment:
we could simplify this method a bit since there's a lot of redundancy here. Some like the following?

PluginResult result;

try {
	JSONObject locationObject = LocationUtils.locationToJSON(locationResult);
	result = new PluginResult(PluginResult.Status.OK, locationObject);
} catch (JSONException e) {
	result = new PluginResult(PluginResult.Status.JSON_EXCEPTION, LocationError.SERIALIZATION_ERROR.toJSON());
}

if (locationContext.getType() == LocationContext.Type.UPDATE) {
    result.setKeepCallback(true);
}
else {
    locationContexts.delete(locationContext.getId());
}

locationContext.getCallbackContext().sendPluginResult(result);

Reply:
I had simply copied the code from the `onLocationResultSuccess` method before. But you're right, we can avoid some code repetition. Already applied your suggestion. Thanks!